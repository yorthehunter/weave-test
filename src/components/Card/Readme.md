Card:

```js
<Card><p>Hi! This is a card. Any content can go in here.</p></Card>
```

Hoverable card:

```js
<Card isHoverable><p>Hi! This is a card. Any content can go in here.</p></Card>
```
