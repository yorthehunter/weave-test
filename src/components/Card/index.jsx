import React from 'react';
import { string, node, bool } from 'prop-types';
import CustomTag from '../utilities/CustomTag';
import { sizeModifier } from '../utilities/util';

const componentPrefix = 'weave-card';

/**
 * Card
 */

const Card = (props) => {
  const {
    isHoverable,
    inset,
    children,
  } = props;

  const componentClasses = [
    componentPrefix,
    sizeModifier(inset, 'u-inset'),
    isHoverable ? 'weave-card--hoverable' : null,
  ];

  return (
    <CustomTag componentClasses={componentClasses} {...props}>
      {children}
    </CustomTag>
  );
};

const defaultProps = {
  tag: 'div',
  isHoverable: false,
  inset: 'md',
  children: false,
};

const propTypes = {
  tag: string,
  isHoverable: bool,
  inset: string,
  children: node,
};

Card.defaultProps = defaultProps;
Card.propTypes = propTypes;
export default Card;
