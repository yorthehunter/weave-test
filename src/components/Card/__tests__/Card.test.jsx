import React from 'react';
import renderer from 'react-test-renderer';
import Card from '../';

describe('Card', () => {
  it('renders correctly with default values', () => {
    const rendered = renderer.create(<Card />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('renders correctly as hoverable', () => {
    const rendered = renderer.create(<Card isHoverable />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
