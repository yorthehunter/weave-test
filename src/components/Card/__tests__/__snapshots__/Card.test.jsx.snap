// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly as hoverable 1`] = `
<div
  className="weave-card u-inset--md weave-card--hoverable"
  href={null}
  onClick={null}
  style={Object {}}
  type={null}
  value={null}
/>
`;

exports[`Card renders correctly with default values 1`] = `
<div
  className="weave-card u-inset--md"
  href={null}
  onClick={null}
  style={Object {}}
  type={null}
  value={null}
/>
`;
